### ISSUE TITLE
Falsey values do not trigger subview's waitFor condition

### ISSUE THREAD

**Issue by bryanspears** [AmpersandJS/ampersand-view#137]
My specific example is a 'number' type where the incoming JSON value is a numeric 0. Doesn't trigger the subview render.

After some testing, this affects ALL falsey values I've tried. So a boolean field with an incoming value of `false` will not trigger the waitFor. Null/undefined, I can understand the reason not to have those function, but boolean false and numeric 0 are very valid use cases.

**Comment by pgilad** [comment]
Yep... this is the problematic line:
https://github.com/AmpersandJS/ampersand-view/blob/master/ampersand-view.js#L301

Should `waitFor` only be triggered for non-null or undefined values or do we have more in the list?

P.S - This is a breaking change

**Comment by kamilogorek** [comment]
Yes, we should probably be just more explicit about those checks.

**Comment by bryanspears** [comment]
This hack has worked for me thus far. Any issues with this? If not, I could whip up a PR soon.
```
waitFor = getPath(this, opts.waitFor);
if (!opts.waitFor || waitFor || (waitFor !== undefined && waitFor !== null)) {
```

**Comment by bryanspears** [comment]
So how would you wait for collections to be populated? If you `waitFor: 'model.collection.length` that would work in the current code, however, what if the collection truly has no models? Any ideas how to correctly waitFor a collection?

**Comment by cdaringe** [comment]
@bryanspears, looks like you had an open PR for this issue.  what was the scoop w/ that?
